libpoolprotocols/stratum/EthStratumClient.cpp

```diff
@@ -741,8 +741,11 @@ void EthStratumClient::processReponse(Json::Value& responseObject)
 
 		case 3:
 
-			// Response to "mining.authorize"
-			m_authorized.store(_isSuccess, std::memory_order_relaxed);
+			// Response to "mining.authorize" (https://en.bitcoin.it/wiki/Stratum_mining_protocol#mining.authorize)
+			// Result should be boolean, some pools also throw an error, so _isSuccess can be false
+			authSuccessful = (!_isSuccess) ? false : ((!jResult.empty() && jResult.isBool()) ? jResult.asBool() : false);
```

Comment:
authSuccessful and below submitSuccesful apply the very same logic over a negation of "_isSuccess".
So if _isSuccess = false (calculated above) this logic makes authSuccessful = false and stop.

Reply:
well for starters i do not have
 ``_isSuccess = !responseObject.get("result", Json::Value::null).empty();``
anymore but
``_isSuccess = responseObject.get("error", Json::Value::null).empty();``

i agree that the authSuccessful and submitSuccessful is a bit long, but the logic behind it is basically
* if error is not null (aka negative method invocation) auth was unsuccessful
* otherwise use the boolean value of "result" field, with fallback to false if not boolean